### ISSUE TITLE
Rscript.exec() returns Try[Unit] which I think is a bad idea

### ISSUE THREAD

**Issue by tfenne** [fulcrumgenomics/fgbio#397]
Rscript has two types of method, `exec()` and `execIfAvailable()`, both of which return `Try[Unit]`.  I think I did this at your behest @nh13 instead of having them simply throw exceptions.

I think this is probably an anti-pattern.  Specifically I think using `Try[Unit]` is bad.  `Try` is nice when returning something the user needs, so that they are forced to deal with calling `recover()` or `getOrElse()` etc.  But when returning Unit, because the method is side-effecting, it's all to easy to ignore the return and return type, and assume the method will throw an exception is something goes wrong.

I just found some code doing exactly that, and think we should migrate those methods to throwing vs. returning Try.  What do you think @nh13?

**Comment by nh13** [comment]
Agreed unless your always using the thing in Try then it doesn’t make to return a Try. Go for it.
